group_project/group_activity.py

```diff
@@ -1,3 +1,75 @@
+"""
+This module parses Group project XML into a tree of domain-specific objects.
+Basically it looks like the following
+
+GroupActivity (.)
+- ActivityComponent (./projectcomponent)
+-- ActivitySection (//section - different paths)
+--- ActivityQuestion (//section/question)
+---- <no dedicated class, etree.Element is used> (//section/question/answer)
+--- ActivityAssessment (//section/assessment)
+---- <no dedicated class, etree.Element is used> (//section/assessment/answer)
+
+GroupActivity (paths relative to root element)
+- resources: [DottableDict(title, description, location)] - ./resources/document
+- grading_criteria: [DottableDict(title, description, location)] - ./resources/document[@grading_criteria="true"]     # attribute filter implicit
+- milestone_dates: {str: datetime.date} - ./dates/milestone
+- submissions: [DottableDict(id, title, description, location?)] - ./submissions/document
+- grade_questions: [ActivityQuestion] - //section/question
```

Comment:
The attributes above all look like they should be moved to the individual stages nodes (or below)